CRM/MembersOnlyEvent/Service/MembersOnlyEventAccessService.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as EventMembershipType;
+use CRM_MembersOnlyEvent_Configurations as Configurations;
+use CRM_MembersOnlyEvent_BAO_MembersOnlyEvent as MembersOnlyEvent;
+
+class CRM_MembersOnlyEvent_Service_MembersOnlyEventAccessService {
+
+  /**
+   * @var array
+   */
+  private $contactActiveAllowedMemberships = [];
+
+  /**
+   * @var \CRM_MembersOnlyEvent_DAO_MembersOnlyEvent|null
+   */
+  private $membersOnlyEvent = NULL;
+
+  /**
+   * @var int|null
+   */
+  private $contactID = NULL;
+
+  /**
+   * @var int|null
+   */
+  private $eventID = NULL;
+
+  public function __construct() {
+    if (!$this->isValidPath()) {
+      return;
+    }
+    $this->eventID = CRM_Utils_Request::retrieve('id', 'Positive');
+
+    $this->membersOnlyEvent = MembersOnlyEvent::getMembersOnlyEvent($this->eventID);
+    $this->contactID = CRM_Core_Session::getLoggedInContactID();
+
+    if ($this->contactID) {
+      // Only calculate for logged users
+      $this->contactActiveAllowedMemberships = $this->getAllowedMemberships($this->membersOnlyEvent, $this->contactID);
+    }
+  }
+
+  /**
+   * Return a list of allowed and active memberships
+   * Also in case 'membership duration check' is enabled.
+   * It check for the membership is valid (active) during the period of
+   * the specified event.
+   *
+   * The membership is valid during an event if :
+   * 1- The membership end date is empty.
+   * 2- The event start date is empty.
```

Comment:
you cannot create an event without a start date, so we don't need such condition